version: "3.8"

services:
  nginx:
    image: julianzh/nginx-devops:latest
    container_name: nginx-devops
    ports:
      - "80:80"
    links:
      - devops
    volumes:
      - ./nginx:/etc/nginx/conf.d:delegated
      - ../:/var/www/html/devops:delegated
    networks:
      devops-network:
        ipv4_address: 172.11.0.3
  devops:
    container_name: devops
    image: julianzh/php-devops:latest
    volumes:
      - ../:/var/www/html/devops:delegated
    networks:
      devops-network:
        ipv4_address: 172.11.0.7
  mysql:
    image: julianzh/mysql-devops:latest
    container_name: devops-db
    restart: "on-failure"
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: devops
    ports:
      - "3306:3306"
    volumes:
      - devops-db:/var/lib/mysql
    networks:
      devops-network:
        ipv4_address: 172.11.0.2
networks:
  devops-network:
    name: devops-network
    driver: bridge
    ipam:
      config:
        - subnet: 172.11.0.0/16
volumes:
  devops-db:
    external: true
